api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -379,25 +379,32 @@ const (
 	PruneRetentionPolicy RetentionPolicy = "Prune"
 )
 
-// +kubebuilder:validation:Enum=ImageRegistry;openshift-samples;Insights
+// +kubebuilder:validation:Enum=ImageRegistry;openshift-samples;Insights;baremetal
 type OptionalCapability string
 
 const ImageRegistryCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityImageRegistry)
-
 const OpenShiftSamplesCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityOpenShiftSamples)
-
 const InsightsCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityInsights)
+const BaremetalCapability OptionalCapability = OptionalCapability(configv1.ClusterVersionCapabilityBaremetal)
 
-// capabilities allows disabling optional components at install time.
+// capabilities allows enabling or disabling optional components at install time.
+// When this is not supplied, the cluster will use the DefaultCapabilitySet defined for the respective
+// OpenShift version, minus the baremetal capability.
 // Once set, it cannot be changed.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.enabled) && has(self.disabled) ? self.enabled.all(e, !(e in self.disabled)) : true", message="Capabilities can not be both enabled and disabled at once."
 type Capabilities struct {
-	// disabled when specified, sets the cluster version baselineCapabilitySet to None
-	// and sets all additionalEnabledCapabilities BUT the ones supplied in disabled.
-	// This effectively disables that capability on the hosted cluster.
-	//
-	// When this is not supplied, the cluster will use the DefaultCapabilitySet defined for the respective
-	// OpenShift version.
+	// enabled when specified, explicitly enables the specified capabilitíes on the hosted cluster.
+	// Once set, this field cannot be changed.
 	//
+	// +listType=atomic
+	// +immutable
+	// +optional
+	// +kubebuilder:validation:MaxItems=25
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Enabled is immutable. Changes might result in unpredictable and disruptive behavior."
+	Enabled []OptionalCapability `json:"enabled,omitempty"`
```

Comment:
the list of valid input for Enabled is different than the one for Disabled. The API should reflect and enforce that so may be we need to different types OptionalCapabilityDisabled / OptionalCapabilityEnabled cc @JoelSpeed

Reply:
Would you consider it an error for a user to force Disable a "Disabled by default" capability? Being explicit that they want it disabled allows them to distinguish between no opinion (the platform can choose) and having an explicit opinion (I definitely don't want this).

By allowing this choice, it allows us in the future to change whether capabilities are enabled/disabled by default without affecting customers who have an opinion.

It may not fit exactly this case, but that's the general rule for why we allow the enable/disable lists to contain the same elements (though not at the same time of course)

Reply:
>By allowing this choice, it allows us in the future to change whether capabilities are enabled/disabled by default without affecting customers who have an opinion.

ok, I can get behind this. Thanks!

In any case let's please add validation for negative cases here TestOnCreateAPIUX

Reply:
I'm trying to understand what you mean: do you mean more test cases in test/e2e/create_cluster_test.go:TestOnCreateAPIUX function?